src/Concordium/Client/Runner.hs

```diff
@@ -3744,3 +4129,43 @@ signEncodedTransaction encPayload sender energy nonce expiry accKeys = Types.Nor
       }
       keys = Map.toList $ fmap Map.toList accKeys
   in Types.signTransaction keys header encPayload
+
+-- |Get the response value of a GRPCResult under the provided mapping.
+-- Returns @Left@ wrapping an error string describing its nature if the
+-- result contains an error, or a @Right@ wrapping the response value
+-- under the provided mapping otherwise.
+getResponseValue :: (a -> b) -> GRPCResult (FromProtoResult a) -> Either String b
+getResponseValue f res =
+  case res of
+    Left err -> Left $ "A GRPC error occurred: " <> err
+    Right resp ->
+      case grpcResponseVal resp of
+        Left _ -> Left "An error was found in the response."
```

Comment:
I believe you mentioned that conversion related error messages should probably not not be visible to the user when we discussed this at some point. What do you think is better here?

Reply:
I agree it should not be visible. But if there is a bug it is better that it is as clear as it can be. 

So propagating the internal error will be useful in a bug report.